### PR TITLE
implement different entry points for operator

### PR DESCRIPTION
**Pull request by pradeepitm12**
this adds two different entry points for operator
- kubernetes
- openshift
both entry points are responsible to add respective extensions
Part of poc moving to knative/pkg

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Comment by pradeepitm12** [comment]
Note : This PR is still in WIP working on the extension part of it.

**Review by vdemeester — COMMENTED** [review]
Any reason for having `tektonpipeline.go`, etc… duplicated ?

**Comment by pradeepitm12** [comment]
> Any reason for having `tektonpipeline.go`, etc… duplicated ?

Yes, it was there to make things simple at first, now we have reduced duplicated code.

File: pkg/reconciler/openshift/tektonpipeline/controller.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2019 The Tekton Authors
```
Comment (savitaashture) [inline pkg/reconciler/openshift/tektonpipeline/controller.go:2]:
```suggestion
Copyright 2020 The Tekton Authors
```

File: pkg/reconciler/openshift/tektonpipeline/extension.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package tektonpipeline
```
Comment (savitaashture) [inline pkg/reconciler/openshift/tektonpipeline/extension.go:1]: Add License here

**Review by savitaashture — COMMENTED** [review]
PR looks good to me
Thank you 

Small nit related to License

File: pkg/reconciler/openshift/tektonpipeline/controller.go (hunk 1 above)
Comment (pradeepitm12) [reply pkg/reconciler/openshift/tektonpipeline/controller.go:2]: That was a miss. Thanks

**Review by savitaashture — COMMENTED** [review]
/lgtm

Thank you :+1:

**Pull request merged**